display-products-via-endpoint

barsha-kc opened the pull request.

barsha-kc committed:
This is my fetch: Barsha

ChrisEST123 commented on Lab3/Fetch.js:
```diff
@@ -0,0 +1,15 @@
+    const data = await response.json();
+
+    document.querySelector(".product-grid").innerHTML = data.products.map(product => `
+    <div class="card">
```
The indenting looks off here. If we want the code to be as readable as possible then it is a good practise to indent the code depending on the logic or code written. Currently everything is on the same line vertically, which makes it look like we are defining several things without them being tied to each other neccessarily. Alos lets make sure that wherever the function ends, the ending is vertically on the same line as well as the function start. You could test around with this and see if it does look more easily readable then. Since we are defining an html inside a map the code would look clearer like this:


```javascript
document.querySelector(".product-grid").innerHTML = data.products.map(product => `
    <div class="card">
        <img src="${product.thumbnail}" alt="${product.title}">
        <h3>${product.title}</h3>
        <p>$${product.price.toFixed(2)}</p>
        <button>Add to Cart</button>
    </div>
`).join("");

ChrisEST123 reviewed (changes requested):
The page itself looks beautiful, as requested, however there is a small issue in the code, which would be nice to have fixed for future maintainability.

As for the branch itself, then a small note. You do not need to write your name into the branch name or the commit/PR message. Via Git we are able to see anyways who did the developments and who not, due to the commits being tied to your user. Let's try to avoid using our names for branch names and messages.

Besides this we should try to have the branch name, commit message and the PR description give over what we are trying to achieve. Since our aim is to obtain data for products in this PR, then a good name for the branch would for example be 'display-products-via-endpoint', commit message would be 'Displaying products on webpage from the endpoint' and description could be 'The aim of this PR is to display products on the web page, with data provided from the products endpoint.' These are just examples but just to give an idea.

If you would like you could make the changes right now or you could just keep this in mind for the next ones. For guidance on how to change the branch name you could refer to this link [here](https://www.ionos.com/digitalguide/websites/web-development/renaming-a-git-branch/#:~:text=Renaming%20a%20remote%20Git%20branch), however it could be quite complex, so do it if you feel adventurous.

barsha-kc committed:
Displaying products on webpage

This aim of this PR is to display products on the web page, with data provided from the products endpoints.

ChrisEST123 reviewed (approved):
Code looks good. Decided to fix branch naming issues with next task. Good to merge
